### PR TITLE
Use RetryableSphereClientWithExponentialBackoff utilities from jvm sdk

### PR DESCRIPTION
**Pull request by salander85**

### CODE REVIEW

File: src/main/java/com/commercetools/project/sync/util/ClientConfigurationUtils.java [hunk 1]
```diff
@@ -29,25 +19,23 @@
...
 
     final HttpClient httpClient = getHttpClient();
-    final SphereAccessTokenSupplier tokenSupplier =
-        SphereAccessTokenSupplier.ofAutoRefresh(clientConfig, httpClient, false);
-    final SphereClient underlying = SphereClient.of(clientConfig, httpClient, tokenSupplier);
-    final SphereClient retryClient = withRetry(underlying);
-    return withLimitedParallelRequests(retryClient);
+    return RetryableSphereClientBuilder.of(clientConfig, httpClient)
+        .withStatusCodesToRetry(
+            Arrays.asList(BAD_GATEWAY_502, SERVICE_UNAVAILABLE_503, GATEWAY_TIMEOUT_504))
```
Comment (ahmetoz) [inline src/main/java/com/commercetools/project/sync/util/ClientConfigurationUtils.java:32]: I think the default is better, which includes 500 too.

File: src/main/java/com/commercetools/project/sync/util/ClientConfigurationUtils.java [hunk 2]
```diff
@@ -29,25 +19,23 @@
...
   }
 
   /**
-   * Gets an asynchronous {@link HttpClient} to be used by the {@link BlockingSphereClient}. Client
-   * is created during first invocation and then cached.
+   * Gets an asynchronous {@link HttpClient} to be used by the {@link SphereClient}. Client is
+   * created during first invocation and then cached.
    *
    * @return {@link HttpClient}
    */
```
Comment (ahmetoz) [inline src/main/java/com/commercetools/project/sync/util/ClientConfigurationUtils.java:41]: I think we could use the same client that we have in the latest java-sync, the defaults in DefaultAsyncHttpClientConfig is good. So DEFAULT_TIMEOUT also not needed.

so 

``` java
/**
     * Gets an asynchronous {@link HttpClient} of `asynchttpclient` library,
     * to be used by as an underlying http client for the {@link SphereClient}.
     *
     * @return an asynchronous {@link HttpClient}
     */
    private static HttpClient getHttpClient() {
        final AsyncHttpClient asyncHttpClient =
            new DefaultAsyncHttpClient(new DefaultAsyncHttpClientConfig.Builder().build());
        return AsyncHttpClientAdapter.of(asyncHttpClient);
    }
```

**Review by ahmetoz — CHANGES_REQUESTED** [review]

File: src/main/java/com/commercetools/project/sync/util/ClientConfigurationUtils.java (hunk 2 above)
Comment (salander85) [reply src/main/java/com/commercetools/project/sync/util/ClientConfigurationUtils.java:31]: c61c772

File: src/main/java/com/commercetools/project/sync/util/ClientConfigurationUtils.java (hunk 1 above)
Comment (salander85) [reply src/main/java/com/commercetools/project/sync/util/ClientConfigurationUtils.java]: c61c772

**Review by ahmetoz — APPROVED** [review]

**Pull request merged**